tests/mpi/hp_cell_weights_03.cc

```diff
@@ -113,6 +113,9 @@ test()
   try
     {
       tria.execute_coarsening_and_refinement();
+      deallog << "Error: we did not throw an exception when we should have"
+              << std::endl;
+      std::abort();
```

Comment:
Can't you just remove the try-catch block and only execute
```
tria.execute_coarsening_and_refinement();
```
?

Reply:
The logic is reversed. I have to catch the exception thrown in `tria.execute_coarsening_and_refinement()` and instead signal an error if it *doesn't* throw.

Reply:
Ah, that makes sense.